platform/qt/qt.cmake

```diff
@@ -5,8 +5,15 @@ option(WITH_QT_DECODERS "Use builtin Qt image decoders" OFF)
 option(WITH_QT_I18N     "Use builtin Qt i18n support"   OFF)
 option(WITH_QT_4        "Use Qt4 instead of Qt5"        OFF)
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fvisibility=hidden -D__QT__")
-set(CMAKE_C_FLAGS   "${CMAKE_C_FLAGS}   -fvisibility=hidden -D__QT__")
+if (CMAKE_COMPILER_IS_GNUCC AND NOT
```

Comment:
This will still disable `-fvisibility=hidden` for clang because it only sets it when `CMAKE_COMPILER_IS_GNUCC`.

I think you could always do `set(USE_DEFAULT_VISIBILITY "-fvisibility=hidden")` and override with `set(USE_DEFAULT_VISIBILITY "")` when the broken compiler is detected.